unittest works builded on 64bits, but failed in 32bits (segfault)

Hi,

build the loader in 64 and 32 bits, unittest works in 64bits but failed (some of them) in 32 bits

the failed test:

~~~
The following tests FAILED:
        366 - UnknownFunction.DeviceFunctionFromGetInstanceProcAddr (SEGFAULT)
        367 - UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer (SEGFAULT)
        372 - UnknownFunction.CommandBufferFunctionFromGetInstanceProcAddr (SEGFAULT)
        373 - UnknownFunction.CommandBufferFunctionFromGetInstanceProcAddrWithImplicitLayer (SEGFAULT)
        376 - UnknownFunction.QueueFunctionFromGetInstanceProcAddr (SEGFAULT)
        377 - UnknownFunction.QueueFunctionFromGetInstanceProcAddrWithImplicitLayer (SEGFAULT)
Errors while running CTest
~~~

if run the test manually, for example, 367,

in 64bits build
~~~
└───╼  tests/test_regression "--gtest_filter=UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer" "--gtest_also_run_disabled_tests" -V
Note: Google Test filter = UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from UnknownFunction
[ RUN      ] UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer
INFO: Vulkan Loader Version 1.3.207
LAYER: Searching for layer manifest files
LAYER:    In following folders:
LAYER:       /etc/vulkan/implicit_layer.d
LAYER:    Found the following files:
LAYER:       /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
INFO: Found manifest file /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json (file version "1.0.0")
LAYER: Searching for layer manifest files
LAYER:    In following folders:
LAYER:       /etc/vulkan/explicit_layer.d
LAYER:    Found no files
DRIVER: Searching for driver manifest files
DRIVER:    In following folders:
DRIVER:       /etc/vulkan/icd.d
DRIVER:    Found the following files:
DRIVER:       /etc/vulkan/icd.d/test_icd_0.json
DRIVER: Found ICD manifest file /etc/vulkan/icd.d/test_icd_0.json, version "1.0.0"
DEBUG: Searching for ICD drivers named /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/icd_manifests/libtest_icd_version_6_0.so
DEBUG: Build ICD instance extension list
LAYER | DEBUG: Loading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | INFO: Insert instance layer VK_LAYER_implicit_layer_unknown_function_intercept (/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so)
LAYER: vkCreateInstance layer callstack setup to:
LAYER:    <Application>
LAYER:      ||
LAYER:    <Loader>
LAYER:      ||
LAYER:    VK_LAYER_implicit_layer_unknown_function_intercept
LAYER:            Type: Implicit
LAYER:                Disable Env Var:  DISABLE_ME
LAYER:            Manifest: /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
LAYER:            Library:  /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER:      ||
LAYER:    <Drivers>

DEBUG: Build ICD instance extension list
LAYER | DEBUG: Loading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | INFO: Inserted device layer VK_LAYER_implicit_layer_unknown_function_intercept (/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so)
LAYER: vkCreateDevice layer callstack setup to:
LAYER:    <Application>
LAYER:      ||
LAYER:    <Loader>
LAYER:      ||
LAYER:    VK_LAYER_implicit_layer_unknown_function_intercept
LAYER:            Type: Implicit
LAYER:                Disable Env Var:  DISABLE_ME
LAYER:            Manifest: /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
LAYER:            Library:  /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER:      ||
LAYER:    <Device>

DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_0
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_1
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_2
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_3
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_4
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_5
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_6
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_7
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_8
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_9
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_10
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_11
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_12
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_13
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_14
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_15
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_16
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_17
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_18
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_19
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_20
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_21
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_22
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_23
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_24
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_25
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_26
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_27
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_28
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_29
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_30
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_31
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_32
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_33
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_34
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_35
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_36
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_37
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_38
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_39
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_40
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_41
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_42
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_43
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_44
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_45
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_46
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_47
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_48
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_49
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_50
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_51
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_52
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_53
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_54
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_55
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_56
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_57
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_58
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_59
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_60
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_61
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_62
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_63
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_64
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_65
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_66
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_67
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_68
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_69
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_70
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_71
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_72
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_73
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_74
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_75
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_76
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_77
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_78
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_79
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_80
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_81
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_82
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_83
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_84
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_85
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_86
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_87
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_88
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_89
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_90
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_91
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_92
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_93
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_94
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_95
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_96
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_97
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_98
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_99
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_100
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_101
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_102
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_103
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_104
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_105
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_106
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_107
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_108
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_109
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_110
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_111
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_112
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_113
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_114
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_115
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_116
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_117
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_118
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_119
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_120
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_121
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_122
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_123
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_124
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_125
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_126
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_127
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_128
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_129
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_130
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_131
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_132
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_133
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_134
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_135
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_136
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_137
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_138
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_139
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_140
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_141
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_142
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_143
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_144
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_145
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_146
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_147
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_148
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_149
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_150
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_151
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_152
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_153
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_154
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_155
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_156
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_157
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_158
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_159
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_160
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_161
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_162
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_163
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_164
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_165
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_166
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_167
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_168
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_169
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_170
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_171
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_172
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_173
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_174
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_175
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_176
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_177
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_178
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_179
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_180
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_181
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_182
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_183
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_184
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_185
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_186
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_187
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_188
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_189
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_190
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_191
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_192
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_193
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_194
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_195
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_196
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_197
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_198
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_199
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_200
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_201
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_202
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_203
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_204
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_205
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_206
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_207
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_208
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_209
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_210
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_211
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_212
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_213
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_214
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_215
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_216
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_217
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_218
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_219
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_220
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_221
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_222
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_223
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_224
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_225
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_226
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_227
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_228
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_229
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_230
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_231
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_232
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_233
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_234
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_235
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_236
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_237
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_238
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_239
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_240
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_241
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_242
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_243
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_244
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_245
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_246
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_247
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_248
DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_249
LAYER | DEBUG: Unloading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | DEBUG: Unloading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build64/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
[       OK ] UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer (12 ms)
[----------] 1 test from UnknownFunction (12 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (12 ms total)
[  PASSED  ] 1 test.
~~~

in 32bits build

~~~
└───╼  tests/test_regression "--gtest_filter=UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer" "--gtest_also_run_disabled_tests" -V
Note: Google Test filter = UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from UnknownFunction
[ RUN      ] UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer
INFO: Vulkan Loader Version 1.3.207
INFO: [Git - Tag: v1.3.207-29-g82e4f4328, Branch/Commit: master]
LAYER: Searching for layer manifest files
LAYER:    In following folders:
LAYER:       /etc/vulkan/implicit_layer.d
LAYER:    Found the following files:
LAYER:       /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
INFO: Found manifest file /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json (file version "1.0.0")
LAYER: Searching for layer manifest files
LAYER:    In following folders:
LAYER:       /etc/vulkan/explicit_layer.d
LAYER:    Found no files
DRIVER: Searching for driver manifest files
DRIVER:    In following folders:
DRIVER:       /etc/vulkan/icd.d
DRIVER:    Found the following files:
DRIVER:       /etc/vulkan/icd.d/test_icd_0.json
DRIVER: Found ICD manifest file /etc/vulkan/icd.d/test_icd_0.json, version "1.0.0"
DEBUG: Searching for ICD drivers named /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/icd_manifests/libtest_icd_version_6_0.so
DEBUG: Build ICD instance extension list
LAYER | DEBUG: Loading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | INFO: Insert instance layer VK_LAYER_implicit_layer_unknown_function_intercept (/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so)
LAYER: vkCreateInstance layer callstack setup to:
LAYER:    <Application>
LAYER:      ||
LAYER:    <Loader>
LAYER:      ||
LAYER:    VK_LAYER_implicit_layer_unknown_function_intercept
LAYER:            Type: Implicit
LAYER:                Disable Env Var:  DISABLE_ME
LAYER:            Manifest: /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
LAYER:            Library:  /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER:      ||
LAYER:    <Drivers>

DEBUG: Build ICD instance extension list
LAYER | DEBUG: Loading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | INFO: Inserted device layer VK_LAYER_implicit_layer_unknown_function_intercept (/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so)
LAYER: vkCreateDevice layer callstack setup to:
LAYER:    <Application>
LAYER:      ||
LAYER:    <Loader>
LAYER:      ||
LAYER:    VK_LAYER_implicit_layer_unknown_function_intercept
LAYER:            Type: Implicit
LAYER:                Disable Env Var:  DISABLE_ME
LAYER:            Manifest: /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
LAYER:            Library:  /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER:      ||
LAYER:    <Device>

DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_0
Violación de segmento (`core' generado)
~~~

gdb is useless in this case due a lack of useful backtrace

~~~
└───╼  gdb
GNU gdb (GDB) 11.2
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word".
(gdb) exec-file tests/test_regression "--gtest_filter=UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer" "--gtest_also_run_disabled_tests" -V
(gdb) run
Starting program: /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/test_regression tests/test_regression "--gtest_filter=UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer" "--gtest_also_run_disabled_tests" -V
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Note: Google Test filter = UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from UnknownFunction
[ RUN      ] UnknownFunction.DeviceFunctionFromGetInstanceProcAddrWithImplicitLayer
INFO: Vulkan Loader Version 1.3.207
INFO: [Git - Tag: v1.3.207-29-g82e4f4328, Branch/Commit: master]
LAYER: Searching for layer manifest files
LAYER:    In following folders:
LAYER:       /etc/vulkan/implicit_layer.d
LAYER:    Found the following files:
LAYER:       /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
INFO: Found manifest file /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json (file version "1.0.0")
LAYER: Searching for layer manifest files
LAYER:    In following folders:
LAYER:       /etc/vulkan/explicit_layer.d
LAYER:    Found no files
DRIVER: Searching for driver manifest files
DRIVER:    In following folders:
DRIVER:       /etc/vulkan/icd.d
DRIVER:    Found the following files:
DRIVER:       /etc/vulkan/icd.d/test_icd_0.json
DRIVER: Found ICD manifest file /etc/vulkan/icd.d/test_icd_0.json, version "1.0.0"
DEBUG: Searching for ICD drivers named /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/icd_manifests/libtest_icd_version_6_0.so
DEBUG: Build ICD instance extension list
LAYER | DEBUG: Loading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | INFO: Insert instance layer VK_LAYER_implicit_layer_unknown_function_intercept (/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so)
LAYER: vkCreateInstance layer callstack setup to:
LAYER:    <Application>
LAYER:      ||
LAYER:    <Loader>
LAYER:      ||
LAYER:    VK_LAYER_implicit_layer_unknown_function_intercept
LAYER:            Type: Implicit
LAYER:                Disable Env Var:  DISABLE_ME
LAYER:            Manifest: /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
LAYER:            Library:  /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER:      ||
LAYER:    <Drivers>

DEBUG: Build ICD instance extension list
LAYER | DEBUG: Loading layer library /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER | INFO: Inserted device layer VK_LAYER_implicit_layer_unknown_function_intercept (/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so)
LAYER: vkCreateDevice layer callstack setup to:
LAYER:    <Application>
LAYER:      ||
LAYER:    <Loader>
LAYER:      ||
LAYER:    VK_LAYER_implicit_layer_unknown_function_intercept
LAYER:            Type: Implicit
LAYER:                Disable Env Var:  DISABLE_ME
LAYER:            Manifest: /etc/vulkan/implicit_layer.d/implicit_layer_unknown_function_intercept.json
LAYER:            Library:  /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Loader/build32/tests/framework/implicit_layer_manifests/VK_LAYER_implicit_layer_unknown_function_intercept_0_libtest_layer_export_version_2.so
LAYER:      ||
LAYER:    <Device>

DEBUG: loader_gpdpa_instance_terminator() unrecognized name vkNotRealFuncTEST_0

Program received signal SIGSEGV, Segmentation fault.
0x00000000 in ?? ()
(gdb) bt
#0  0x00000000 in ?? ()
#1  0x56747194 in ?? ()
#2  0x567984fa in ?? ()
#3  0x5678aaf3 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)
(gdb) bt full
#0  0x00000000 in ?? ()
No symbol table info available.
#1  0x56747194 in ?? ()
No symbol table info available.
#2  0x567984fa in ?? ()
No symbol table info available.
#3  0x5678aaf3 in ?? ()
No symbol table info available.
Backtrace stopped: previous frame inner to this frame (corrupt stack?)
quit) 
A debugging session is active.

        Inferior 1 [process 51881] will be killed.

Quit anyway? (y or n) y
~~~

archlinux
gcc 11.2.0
Vulkan-Loader builded with `-DUPDATE_DEPS=ON`

revert all and sync with KhronosGroup/Vulkan-Loader master branch 

unitest failed in all 32bits test (except the mentioned in https://github.com/KhronosGroup/Vulkan-Loader/issues/909 with

~~~
388/409 Test #388: UnknownFunctionDeathTests.PhysicalDeviceFunctionErrorPath .......................***Failed    0.14 sec
Note: Google Test filter = UnknownFunctionDeathTests.PhysicalDeviceFunctionErrorPath
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from UnknownFunctionDeathTests
[ RUN      ] UnknownFunctionDeathTests.PhysicalDeviceFunctionErrorPath
INFO:             Vulkan Loader Version 1.3.211
INFO:             [Git - Tag: bd3fe8e, Branch/Commit: master]
LAYER:            Searching for layer manifest files
LAYER:               In following folders:
LAYER:                  /etc/vulkan/implicit_layer.d
LAYER:                  /usr/local/etc/vulkan/implicit_layer.d
LAYER:               Found no files
LAYER:            Searching for layer manifest files
LAYER:               In following folders:
LAYER:                  /etc/vulkan/explicit_layer.d
LAYER:                  /usr/local/etc/vulkan/explicit_layer.d
LAYER:               Found no files
DRIVER:           Searching for driver manifest files
DRIVER:              In following folders:
DRIVER:                 /etc/vulkan/icd.d
DRIVER:                 /usr/local/etc/vulkan/icd.d
DRIVER:              Found the following files:
DRIVER:                 /usr/local/etc/vulkan/icd.d/test_icd_1.json
DRIVER:                 /usr/local/etc/vulkan/icd.d/test_icd_0.json
DRIVER:           Found ICD manifest file /usr/local/etc/vulkan/icd.d/test_icd_1.json, version "1.0.0"
DEBUG:            Searching for ICD drivers named /home/runner/work/Vulkan-Loader/Vulkan-Loader/build/tests/framework/icd_manifests/libtest_icd_version_2_export_icd_gpdpa_1.so
DRIVER:           Found ICD manifest file /usr/local/etc/vulkan/icd.d/test_icd_0.json, version "1.0.0"
DEBUG:            Searching for ICD drivers named /home/runner/work/Vulkan-Loader/Vulkan-Loader/build/tests/framework/icd_manifests/libtest_icd_version_2_export_icd_gpdpa_0.so
DEBUG:            Build ICD instance extension list
LAYER:            vkCreateInstance layer callstack setup to:
LAYER:               <Application>
LAYER:                 ||
LAYER:               <Loader>
LAYER:                 ||
LAYER:               <Drivers>

DEBUG:            Build ICD instance extension list
DEBUG:            Build ICD instance extension list
DEBUG:            loader_phys_dev_ext_gpa: Adding unknown physical function vkNotIntRealFuncTEST_0 to internal store at index 0
DEBUG:            loader_phys_dev_ext_gpa: Driver /home/runner/work/Vulkan-Loader/Vulkan-Loader/build/tests/framework/icd_manifests/libtest_icd_version_2_export_icd_gpdpa_0.so returned ptr 0xf35991ea for vkNotIntRealFuncTEST_0
/home/runner/work/Vulkan-Loader/Vulkan-Loader/tests/loader_unknown_ext_tests.cpp:205: Failure
Death test: returned_func_i(phys_dev_to_use, 0)
    Result: died but not with expected error.
  Expected: contains regular expression "Extension vkNotIntRealFuncTEST_0 not supported for this physical device"
Actual msg:
[  DEATH   ] AddressSanitizer:DEADLYSIGNAL
[  DEATH   ] =================================================================
[  DEATH   ] ==5073==ERROR: AddressSanitizer: SEGV on unknown address 0x83fffbe8 (pc 0xf359921a bp 0xfff00d28 sp 0xffefffe6 T0)
[  DEATH   ] ==5073==The signal is caused by a WRITE memory access.
[  DEATH   ]     #0 0xf359921a  /home/runner/work/Vulkan-Loader/Vulkan-Loader/loader/unknown_ext_chain_gas_x86.S:387
[  DEATH   ] 
[  DEATH   ] AddressSanitizer can not provide additional info.
[  DEATH   ] SUMMARY: AddressSanitizer: SEGV /home/runner/work/Vulkan-Loader/Vulkan-Loader/loader/unknown_ext_chain_gas_x86.S:387 
[  DEATH   ] ==5073==ABORTING
[  DEATH   ] 
[  FAILED  ] UnknownFunctionDeathTests.PhysicalDeviceFunctionErrorPath (85 ms)
[----------] 1 test from UnknownFunctionDeathTests (85 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (85 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] UnknownFunctionDeathTests.PhysicalDeviceFunctionErrorPath

 1 FAILED TEST
~~~

~~~
The following tests FAILED:
	388 - UnknownFunctionDeathTests.PhysicalDeviceFunctionErrorPath (Failed)
Errors while running CTest
Error: Process completed with exit code 8.
~~~

https://github.com/sl1pkn07/Vulkan-Loader/runs/5872512671?check_suite_focus=true

--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -222,6 +222,15 @@ else() # i.e.: Linux
         endif()
     endif()
 
+    # When compiling for x86 on x64, we can't use CMAKE_SYSTEM_PROCESSOR to determine which architecture to use,
+    # Instead, check the size of void* and if its 4, set ASM_OFFSET_SYSTEM_PROCESSOR to x86
+    # Note - there is no 32 bit arm assembly code, so this only applies to x86 currently.
+    if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+        set(ASM_OFFSET_SYSTEM_PROCESSOR ${CMAKE_SYSTEM_PROCESSOR}) # x86_64 or aarch64
+    else()
+        set(ASM_OFFSET_SYSTEM_PROCESSOR "x86")
+    endif()
+
     if(ASSEMBLER_WORKS)
         add_executable(asm_offset asm_offset.c)
         target_link_libraries(asm_offset loader_specific_options)
@@ -243,14 +252,14 @@ else() # i.e.: Linux
             # Run parse_asm_values.py on asm_offset's assembly file to generate the gen_defines.asm, which the asm code depends on
             add_custom_command(TARGET asm_offset POST_BUILD
                 COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/scripts/parse_asm_values.py "$<TARGET_FILE_DIR:asm_offset>/gen_defines.asm"
-                    "${ASM_OFFSET_INTERMEDIATE_LOCATION}" "GAS" "${CMAKE_CXX_COMPILER_ID}" "${CMAKE_SYSTEM_PROCESSOR}"
+                    "${ASM_OFFSET_INTERMEDIATE_LOCATION}" "GAS" "${CMAKE_CXX_COMPILER_ID}" "${ASM_OFFSET_SYSTEM_PROCESSOR}"
                 BYPRODUCTS gen_defines.asm
                 )
         endif()
         add_custom_target(loader_asm_gen_files DEPENDS gen_defines.asm)
     else()
         if(USE_GAS)
-            message(WARNING "Could not find working ${CMAKE_SYSTEM_PROCESSOR} GAS assembler\n${ASM_FAILURE_MSG}")
+            message(WARNING "Could not find working ${ASM_OFFSET_SYSTEM_PROCESSOR} GAS assembler\n${ASM_FAILURE_MSG}")
         else()
             message(WARNING "Assembly sources have been disabled\n${ASM_FAILURE_MSG}")
         endif()

--- a/loader/log.c
+++ b/loader/log.c
@@ -229,3 +229,8 @@ void loader_log(const struct loader_instance *inst, VkFlags msg_type, int32_t ms
         assert(false);
     }
 }
+
+void loader_log_asm_function_not_supported(const struct loader_instance *inst, VkFlags msg_type, int32_t msg_code,
+                                           const char *func_name) {
+    loader_log(inst, msg_type, msg_code, "Function %s not supported for this physical device", func_name);
+}

--- a/loader/log.h
+++ b/loader/log.h
@@ -49,3 +49,8 @@ uint32_t loader_get_debug_level(void);
 // Logs a message to stderr
 // May output to DebugUtils if the instance isn't null and the extension is enabled.
 void loader_log(const struct loader_instance *inst, VkFlags msg_type, int32_t msg_code, const char *format, ...);
+
+// Used for the assembly code to emit an specific error message
+// This is a work around for linux 32 bit error handling not passing relocatable strings correctly
+void loader_log_asm_function_not_supported(const struct loader_instance *inst, VkFlags msg_type, int32_t msg_code,
+                                           const char *func_name);

--- a/loader/unknown_ext_chain.c
+++ b/loader/unknown_ext_chain.c
@@ -48,7 +48,7 @@
         struct loader_icd_term *icd_term = phys_dev_term->this_icd_term;                                           \
         struct loader_instance *inst = (struct loader_instance *)icd_term->this_instance;                          \
         if (NULL == icd_term->phys_dev_ext[num]) {                                                                 \
-            loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0, "Extension %s not supported for this physical device",    \
+            loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0, "Function %s not supported for this physical device",    \
                        inst->phys_dev_ext_disp_functions[num]);                                                    \
         }                                                                                                          \
         icd_term->phys_dev_ext[num](phys_dev_term->phys_dev);                                                      \

--- a/loader/unknown_ext_chain_gas_aarch64.S
+++ b/loader/unknown_ext_chain_gas_aarch64.S
@@ -76,7 +76,7 @@ vkdev_ext\num:
 .data
 
 termin_error_string:
-.string "Extension %s not supported for this physical device"
+.string "Function %s not supported for this physical device"
 
 .text
 

--- a/loader/unknown_ext_chain_gas_x86.S
+++ b/loader/unknown_ext_chain_gas_x86.S
@@ -100,12 +100,11 @@ vkPhysDevExtTermin\num:
     jmp     [eax + (DISPATCH_OFFSET_ICD_TERM + (PTR_SIZE * \num))]              # Jump to the next function in the chain
 terminError\num:
     mov     eax, dword ptr [eax + INSTANCE_OFFSET_ICD_TERM]                       # Load the loader_instance into eax
-    push    dword ptr [eax + (FUNCTION_OFFSET_INSTANCE + (CHAR_PTR_SIZE * \num))] # Push the func name (fifth arg)
-    push    offset termin_error_string@GOT                                        # Push the error string (fourth arg)
+    push    dword ptr [eax + (FUNCTION_OFFSET_INSTANCE + (CHAR_PTR_SIZE * \num))] # Push the func name (fourth arg)
     push    0                                                                     # Push zero (third arg)
     push    VULKAN_LOADER_ERROR_BIT                                               # Push the error logging bit (second arg)
     push    eax                                                                   # Push the loader_instance (first arg)
-    call    loader_log                                                            # Log the error message before we crash
+    call    loader_log_asm_function_not_supported                                 # Log the error message before we crash
     add     esp, 20                                                               # Clean up the args
     mov     eax, 0
     jmp     eax                                                         # Crash intentionally by jumping to address zero
@@ -129,7 +128,7 @@ vkdev_ext\num:
 .data
 
 termin_error_string:
-.string "Extension %s not supported for this physical device"
+.string "Function %s not supported for this physical device"
 
 .text
 

--- a/loader/unknown_ext_chain_masm.asm
+++ b/loader/unknown_ext_chain_masm.asm
@@ -119,7 +119,7 @@ endm
 ENDIF
 
 .const
-    termin_error_string db 'Extension %s not supported for this physical device', 0
+    termin_error_string db 'Function %s not supported for this physical device', 0
 
 .code
 

--- a/scripts/parse_asm_values.py
+++ b/scripts/parse_asm_values.py
@@ -35,7 +35,8 @@ source_asm_file = sys.argv[2]
 assembler_type = sys.argv[3]
 # Whether we are using gcc, clang, or msvc
 compiler = sys.argv[4]
-# taken from CMAKE_SYSTEM_PROCESSOR - x86_64 or aarch64
+# taken from CMAKE_SYSTEM_PROCESSOR - x86_64, aarch64, or x86
+# Only used with GAS - MASM doesn't need this, as it has its own way to determine x86 vs x64
 arch = sys.argv[5]
 
 if destination_file is None or source_asm_file is None or assembler_type is None or compiler is None or arch is None:
@@ -71,6 +72,7 @@ with open(destination_file, "w", encoding="utf-8") as dest:
             dest.write(".set X86_64, 1\n")
         elif arch == "aarch64":
             dest.write(".set AARCH_64, 1\n")
+        # Nothing to write in the x86 case
 
     for d in defines:
         match = None

--- a/tests/loader_unknown_ext_tests.cpp
+++ b/tests/loader_unknown_ext_tests.cpp
@@ -445,7 +445,7 @@ TEST(UnknownFunctionDeathTests, PhysicalDeviceFunctionErrorPath) {
     decltype(custom_physical_device_functions::func_zero)* returned_func_i =
         env.vulkan_functions.load(inst.inst, function_names.at(0).c_str());
     ASSERT_NE(returned_func_i, nullptr);
-    ASSERT_DEATH(returned_func_i(phys_dev_to_use, 0), "Extension vkNotIntRealFuncTEST_0 not supported for this physical device");
+    ASSERT_DEATH(returned_func_i(phys_dev_to_use, 0), "Function vkNotIntRealFuncTEST_0 not supported for this physical device");
 }
 
 TEST(UnknownFunction, PhysicalDeviceFunctionWithImplicitLayerImplementation) {

